src/main/java/ca/uhn/fhir/jpa/starter/CustomConsentService.java

```diff
@@ -0,0 +1,108 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.util.List;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Base;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.stereotype.Service;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import ca.uhn.fhir.rest.api.RequestTypeEnum;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.interceptor.consent.ConsentOutcome;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentContextServices;
+import ca.uhn.fhir.rest.server.interceptor.consent.IConsentService;
+
+@Service
+public class CustomConsentService implements IConsentService {
+
+  private static final String OAUTH_CLAIM_NAME = System.getenv("OAUTH_CLAIM_NAME");
+
+  private OAuth2Helper oAuth2Helper = new OAuth2Helper();
+
+  @Override
+  public ConsentOutcome startOperation(RequestDetails theRequestDetails,
+      IConsentContextServices theContextServices) {
+    RequestTypeEnum requestType = theRequestDetails.getRequestType();
+    if (requestType.equals(RequestTypeEnum.POST) || requestType.equals(RequestTypeEnum.PUT)) {
+      return validateRequest(theRequestDetails);
+    } else if (requestType.equals(RequestTypeEnum.GET)
+        || requestType.equals(RequestTypeEnum.DELETE)) {
+      return ConsentOutcome.PROCEED;
+    }
+    return ConsentOutcome.REJECT;
```

Comment:
I'm confused by the code in this method. My reading of the doc suggests that the only thing we should be doing in this method is determining if the patient claim is present or not.

i.e. (psuedo code)
```
if (isPatientClaimPresent(theRequestDetails) {
  return ConsentOutcome.PROCEED;
}
return ConsentOutcome.AUTHORIZED;
```